experimental/InvariantTheory/src/TorusInvariantsFast.jl

```diff
@@ -96,12 +187,84 @@ mutable struct InvariantRingFastTorus
     end
 end
 
-invariant_ring(R::RepresentationReductiveGroupFastTorus) = InvariantRingFastTorus(R)
-poly_ring(R::InvariantRingFastTorus) = R.poly_ring
-group(R::InvariantRingFastTorus) = R.group
-representation(R::InvariantRingFastTorus) = R.representation
+@doc raw"""
+    invariant_ring(r::RepresentationTorusGroup)
+
+Return the invariant ring of the torus group represented by `r`.
+
+!!! note
+    The creation of invariant rings is lazy in the sense that no explicit computations are done until specifically invoked (for example, by the `fundamental_invariants` function).
+
+# Examples
+```jldoctest
+julia> T = torus_group(QQ,2);
 
-function fundamental_invariants(z::InvariantRingFastTorus)
+julia> r = representation_from_weights(T, [-1 1; -1 1; 2 -2; 0 -1]);
+
+julia> RT = invariant_ring(r)
+Invariant Ring of
+graded multivariate polynomial ring in 4 variables over QQ under group action of torus of rank2
+```
+"""
+invariant_ring(R::RepresentationTorusGroup) = TorGrpInvRing(R)
+
+@doc raw"""
+    poly_ring(RT::TorGrpInvRing)
+
+# Examples
+```jldoctest
+julia> T = torus_group(QQ,2)
+Torus of rank 2
+  over QQ
+```
+"""
+poly_ring(R::TorGrpInvRing) = R.poly_ring
```

Comment:
For the invariants of finite groups, we call this `polynomial_ring` by now. (This is not really a flaw with this pull request.)